backend/dal/Helpers/Extensions/PropertyExtensions.cs

```diff
@@ -146,6 +147,146 @@ public static IQueryable<Entity.Views.Property> GenerateQuery(this PimsContext c
             else
                 query = query.OrderBy(p => p.AgencyCode).ThenBy(p => p.PID).ThenBy(p => p.PIN).ThenBy(p => p.PropertyTypeId);
 
+            return query;
+        }
+
+
+        /// <summary>
+        /// Generate a query for the specified 'filter'.
+        /// Returns all properties, even properties that don't belong to the user's agency or sub-agencies.
+        /// The results of this query must be 'cleaned' so that only appropriate data is returned to API.
+        /// </summary>
+        /// <param name="context"></param>
+        /// <param name="user"></param>
+        /// <param name="filter"></param>
+        /// <returns></returns>
+        public static IQueryable<Entity.Views.Property> GenerateAllPropertyQuery(this PimsContext context, ClaimsPrincipal user, Entity.Models.AllPropertyFilter filter)
+        {
+            filter.ThrowIfNull(nameof(filter));
+            filter.ThrowIfNull(nameof(user));
+
+            // Check if user has the ability to view sensitive properties.
+            var userAgencies = user.GetAgenciesAsNullable();
+            var isAdmin = user.HasPermission(Permissions.AdminProperties);
+            var viewSensitive = user.HasPermission(Permissions.SensitiveView);
+
+            // Users may only view sensitive properties if they have the `sensitive-view` claim and belong to the owning agency.
+            var query = context.Properties
+                .AsNoTracking()
+                .Where(p => p.ClassificationId != 4); // Disposed properties are not visible.
+
+            // Users are not allowed to view sensitive properties outside of their agency or sub-agencies.
+            if (!isAdmin)
+                query = query.Where(p => !p.IsSensitive || (p.IsSensitive && userAgencies.Contains(p.AgencyId)));
+
+            // Only return properties owned by user's agency or sub-agencies.
+            if (!filter.IncludeAllProperties)
```

Comment:
Here's the magic new part that allows finding properties owned by other agencies.

Reply:
It however has an unintended consequence of essentially allowing other agencies to know who owns what.  Which is what SRES wants to avoid, as they want to be called instead of the owning agency...

Reply:
Could do a weird join with an inner table where you replace the actual agency with SRES maybe?